coderd/inboxnotifications.go

```diff
@@ -28,6 +29,45 @@ const (
 	notificationFormatPlaintext = "plaintext"
 )
 
+const (
+	FallbackIconWorkspace = "DEFAULT_WORKSPACE_ICON"
+	FallbackIconAccount   = "DEFAULT_ACCOUNT_ICON"
+	FallbackIconTemplate  = "DEFAULT_TEMPLATE_ICON"
+	FallbackIconOther     = "DEFAULT_OTHER_ICON"
+)
+
+var fallbackIcons = map[uuid.UUID]string{
+	// workspace related notifications
+	notifications.TemplateWorkspaceCreated:           FallbackIconWorkspace,
+	notifications.TemplateWorkspaceCreated:           FallbackIconWorkspace,
+	notifications.TemplateWorkspaceManuallyUpdated:   FallbackIconWorkspace,
+	notifications.TemplateWorkspaceDeleted:           FallbackIconWorkspace,
+	notifications.TemplateWorkspaceAutobuildFailed:   FallbackIconWorkspace,
+	notifications.TemplateWorkspaceDormant:           FallbackIconWorkspace,
+	notifications.TemplateWorkspaceAutoUpdated:       FallbackIconWorkspace,
+	notifications.TemplateWorkspaceMarkedForDeletion: FallbackIconWorkspace,
+	notifications.TemplateWorkspaceManualBuildFailed: FallbackIconWorkspace,
+	notifications.TemplateWorkspaceOutOfMemory:       FallbackIconWorkspace,
+	notifications.TemplateWorkspaceOutOfDisk:         FallbackIconWorkspace,
+
+	// account related notifications
+	notifications.TemplateUserAccountCreated:           FallbackIconAccount,
+	notifications.TemplateUserAccountDeleted:           FallbackIconAccount,
+	notifications.TemplateUserAccountSuspended:         FallbackIconAccount,
+	notifications.TemplateUserAccountActivated:         FallbackIconAccount,
+	notifications.TemplateYourAccountSuspended:         FallbackIconAccount,
+	notifications.TemplateYourAccountActivated:         FallbackIconAccount,
+	notifications.TemplateUserRequestedOneTimePasscode: FallbackIconAccount,
+
+	// template related notifications
+	notifications.TemplateTemplateDeleted:             FallbackIconTemplate,
+	notifications.TemplateTemplateDeprecated:          FallbackIconTemplate,
+	notifications.TemplateWorkspaceBuildsFailedReport: FallbackIconTemplate,
+
+	// other related notifications
+	notifications.TemplateTestNotification: FallbackIconOther,
```

Comment:
WDYT about returning `FallbackIconOther` if `fallbackIcons[notif.TemplateID]` does not exist? so eventually the order would be:

1. Custom notif icon.
2. `fallbackIcons[notif.TemplateID]`
3. `FallbackIconOther`

In this case we won't need to add entries to fallbackIcons if FallbackIconOther is fine, including `TemplateTestNotification`.

Reply:
Thanks for the idea, like it and will make it more reliable. I added a fallback logic, please tell me if that's what you had in mind.

Reply:
Yes, thank you!